tests/e2e/swarm_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test suite includes test cases to verify basic vDVS functionality
+// in docker swarm mode.
+
+package e2e
+
+import (
+	"log"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+)
+
+type SwarmTestSuite struct {
+	esxName     string
+	master      string
+	worker1     string
+	worker2     string
+	dockerHosts []string
+	volumeName  string
+	serviceName string
+}
+
+func (s *SwarmTestSuite) SetUpTest(c *C) {
+	s.esxName = inputparams.GetEsxIP()
+	s.master = inputparams.GetSwarmManager1()
+	s.worker1 = inputparams.GetSwarmWorker1()
+	s.worker2 = inputparams.GetSwarmWorker2()
+	s.dockerHosts = []string{s.master, s.worker1, s.worker2}
```

Comment:
if you think these vars need to be read just once for the entire test suite, you can have them in SetUpSuite instead of SetupTest.